markdown: support link reference

gemmaro opened the pull request:
This pull request marks link reference definitions in the markdown document as no-wrap, which avoids unintentional line breaks at the colon in the long link reference definition line.

Here are some limitations below, but I suspect it works in most cases:

* Multiline link reference definition is not supported ([Example 193](https://spec.commonmark.org/0.30/#example-193))
* It misinterprets a link reference definition if its link text is invalid ([Example 197](https://spec.commonmark.org/0.30/#example-197))
* Link reference definition may interrupt normal paragraph ([Example 213](https://spec.commonmark.org/0.30/#example-213))

mquinson commented:
Hello @gemmaro, thanks for this PR. I see that it's marked as a draft, so I'll refrain from merging it, but it sounds pretty good to me. I'd love a little note in the NEWS file also, but I can do that myself afterward if you prefer.

What do you plan to do about it before unmarking it 'draft'?

Thanks again,

gemmaro commented:
I updated regexp pattern, to accept indentation for example, according to the CommonMark specification.
I also applied tidyall to format the code, and added about the change to the NEWS file.
